test/e2e/Tests/Helpers/DurableHelpers.cs

```diff
@@ -31,28 +31,39 @@ public OrchestrationStatusDetails(string statusQueryResponse)
         }
     }
 
-    internal static string ParseStatusQueryGetUri(HttpResponseMessage invocationStartResponse)
+    private static string TokenizeAndGetValueFromKeyAsString(string? json, string key)
     {
-        string? responseString = invocationStartResponse.Content?.ReadAsStringAsync().Result;
-
-        if (string.IsNullOrEmpty(responseString))
+        if (string.IsNullOrEmpty(json))
         {
             return string.Empty;
         }
-        JsonNode? responseJsonNode = JsonNode.Parse(responseString);
+        JsonNode? responseJsonNode = JsonNode.Parse(json);
         if (responseJsonNode == null)
         {
             return string.Empty;
         }
 
-        string? statusQueryGetUri = responseJsonNode["StatusQueryGetUri"]?.GetValue<string>();
+        string? statusQueryGetUri = responseJsonNode[key]?.GetValue<string>();
         return statusQueryGetUri ?? string.Empty;
     }
-    internal static OrchestrationStatusDetails GetRunningOrchestrationDetails(string statusQueryGetUri)
+
+    internal static async Task<string> ParseStatusQueryGetUri(HttpResponseMessage invocationStartResponse)
+    {
+        string? responseString = await invocationStartResponse.Content.ReadAsStringAsync();
+        return TokenizeAndGetValueFromKeyAsString(responseString, "StatusQueryGetUri");
+    }
+
+    internal static async Task<string> ParseInstanceId(HttpResponseMessage invocationStartResponse)
```

Comment:
Consider suffixing these two method names with `Async`, which is the best practice in .NET for methods that return a `Task`.

Reply:
Done